### PR TITLE
Code cleanup for relecov_core

### PR DESCRIPTION
**Pull request by Shettland**
This PR addresses several changes to the relecov_core code:

- Renamed files in relecov_core which had redundant names `samples_handling.py` -> `samples.py`
- Imports in relecov_core and relecov_dashboard have been updated accordingly to these new filenames
- Changed local `from` imports to include the whole library name in the import, in order to detect where the imported object come from in the code.
- All the imported functions have been renamed accordingly
- Included some #FIXME and #TODO tags in the code with comments to review the code

### CODE REVIEW

File: relecov_core/api/utils/variants.py [hunk 1]
```diff
@@ -1,3 +1,6 @@
+import relecov_core.models
+import relecov_core.utils.variants
+
 from relecov_core.api.serializers import (
```
Comment (Daniel-VM) [inline relecov_core/api/utils/variants.py]: I think local imports should follow a consisten structure, that is, importing the file and accessing functions when they are invoked in the code  (as in the first two imports). I would prefer to do it this way:

```
import relecov_core.models
import relecov_core.utils.variants
import relecov_core.api.serializers
import relecov_core.core_config

[...]

effect_serializer = relecov_core.api.serializers.CreateEffectSerializer(data={"effect": effect_value})
```

What do you think?

File: relecov_core/api/views.py [hunk 2]
```diff
@@ -1,3 +1,4 @@
+import relecov_core.urls
```
Comment (Daniel-VM) [inline relecov_core/api/views.py]: Imports can be organized by separating generic libraries from local imports. See below:

```
# Generic imports
from django.contrib.auth.decorators import login_required
from django.shortcuts import render

# Local imports
import relecov_core.core_config
import relecov_core.utils.handling_lineage
import relecov_core.utils.handling_variant
import relecov_dashboard.dashboard_config
import relecov_dashboard.utils.met_bioinfo
import relecov_dashboard.utils.met_host_info
```

> if this suggestion is approved, extend it to all py files in `relecov_core`

File: relecov_core/utils/lineage.py [hunk 3]
```diff
@@ -1,11 +1,11 @@
-from relecov_core.utils.handling_samples import get_sample_obj_from_id
-from relecov_core.models import LineageFields, LineageValues
+from relecov_core.utils.samples import get_sample_obj_from_id
```
Comment (Daniel-VM) [inline relecov_core/utils/lineage.py]: missing import refactor

File: relecov_core/utils/metadata.py [hunk 4]
```diff
@@ -3,8 +3,6 @@
 from pathlib import Path
 
 from relecov_core.core_config import METADATA_UPLOAD_FOLDER
-
-# from relecov_core.utils.generic_functions import store_file
 from relecov_core.models import Document
```
Comment (Daniel-VM) [inline relecov_core/utils/metadata.py]: missing import refactor

File: relecov_core/utils/plotly_dash_graphics.py [hunk 5]
```diff
@@ -1,20 +1,17 @@
 from django_plotly_dash import DjangoDash
 from dash.dependencies import Input, Output
 from dash import dcc, html
-import plotly.express as px
```
Comment (Daniel-VM) [inline relecov_core/utils/plotly_dash_graphics.py:4]: plotly.express is tipically imported as `px` (https://plotly.com/python/plotly-express/). Following this 'convention' can improve readability.

File: relecov_core/utils/plotly_graphics.py [hunk 6]
```diff
@@ -1,11 +1,8 @@
 from plotly.offline import plot
-
-import plotly.graph_objects as go
```
Comment (Daniel-VM) [inline relecov_core/utils/plotly_graphics.py:3]: I would also revert this change to follow 'convention' for plotly.graph_objects (as `go`) and plotly.figure_factory (as ff).

File: relecov_core/utils/samples.py [hunk 7]
```diff
@@ -3,57 +3,20 @@
...
-    # save_sample_form_data,
-)
-
-from relecov_core.utils.generic_functions import get_configuration_value
-
-from relecov_core.utils.plotly_dash_graphics import dash_bar_lab
+import relecov_tools.utils
+
+import relecov_core.utils.plotly_dash_graphics
+import relecov_core.core_config as core_config
```
Comment (Daniel-VM) [inline relecov_core/utils/samples.py]: We should reach a consensus on whether to use aliases or raw import names to ensure consistency across all files.

File: relecov_core/utils/samples_graphics.py [hunk 8]
```diff
@@ -1,15 +1,11 @@
-from relecov_core.utils.plotly_graphics import (
-    line_graphic,
-    bar_graphic,
-)
+import relecov_core.utils.plotly_graphics
+import relecov_core.utils.rest_api_handling
+from relecov_core.utils.samples import get_all_recieved_samples_with_dates
```
Comment (Daniel-VM) [inline relecov_core/utils/samples_graphics.py]: missing import refactoring?

File: relecov_core/utils/variants.py [hunk 9]
```diff
@@ -1,44 +1,30 @@
...
-    get_sample_obj_from_sample_name,
-)
-
-from relecov_core.core_config import HEADING_FOR_VARIANT_TABLE_DISPLAY
-
-
-from relecov_core.utils.plotly_graphics import needle_plot
+import relecov_core.models
+import relecov_core.core_config
+import relecov_core.utils.samples as samples
```
Comment (Daniel-VM) [inline relecov_core/utils/variants.py]: another alias found.

File: relecov_core/views.py [hunk 10]
```diff
@@ -371,15 +370,17 @@ def intranet(request):
 
 
 def variants(request):
-    return render(request, "relecov_core/variants.html", {})
+    return render(request, "relecov_core/relecov_core.utils.variants.html", {})
```
Comment (Daniel-VM) [inline relecov_core/views.py]: is it a mistake?

File: relecov_dashboard/utils/graphics/variant_mutations_in_lineages_mutation_table.py [hunk 11]
```diff
@@ -13,7 +13,8 @@
 from django_plotly_dash import DjangoDash
 from dash import dash_table
 from relecov_core.models import Effect, Gene, VariantAnnotation, VariantInSample
```
Comment (Daniel-VM) [inline relecov_dashboard/utils/graphics/variant_mutations_in_lineages_mutation_table.py]: missing import refactoring?

File: relecov_dashboard/utils/methodology_index.py [hunk 12]
```diff
@@ -1,6 +1,6 @@
 from statistics import mean
 
-from relecov_core.utils.handling_bioinfo_analysis import (
+from relecov_core.utils.bioinfo_analysis import (
```
Comment (Daniel-VM) [inline relecov_dashboard/utils/methodology_index.py]: This should be `import relecov_core.utils.bioinfo_analysis` to ensure consistency.

**Review by Daniel-VM — COMMENTED** [review]
It looks good. Just few things to accomplish: 

- There are a few missing refactorings of imports that need to be addressed globally in the PR.
- Local imports should follow a consistent structure accross files by importing the entire file rather than specific components.
- Follow plotly conventions when importing this library.
- There are aliases  used inconsistently in local imports. We need to decide whether to use them for local imports or not.

File: relecov_core/api/utils/variants.py (hunk 1 above)
Comment (saramonzon) [reply relecov_core/api/utils/variants.py]: yes, it should be like @Daniel-VM says

File: relecov_core/utils/samples.py (hunk 7 above)
Comment (saramonzon) [reply relecov_core/utils/samples.py]: I prefer not to use aliases, in this case you can rename core_config as only config as you will call relecov_core.config, it'd be self-explained

We need to remove relecov_ from the module names also, but it's soooo "peñazo"

**Comment by saramonzon** [comment]
I see some renaming left, where p.e "handling" can be removed from the file name

**Review by Daniel-VM — APPROVED** [review]
LGTM
